### PR TITLE
refactor: update stack with component-rollup-config

### PR DESCRIPTION
**Pull request by cbourgois**
### Description of the Change

* use `@ovh-ux/component-rollup-config` to build the package
* remove use of bower
* clean dependencies
* update eslint config

BREAKING CHANGE: the package is now `@ovh-ux/ng-ovh-contact`. The module
is now `ngOvhContact`

e9da2ec — refactor: update stack with component-rollup-config

# To Do : 

* [x] rename repository to `ng-ovh-contact`

https://github.com/ovh-ux/manager/issues/196

### CODE REVIEW

**Review by antleblanc — CHANGES_REQUESTED** [review]
You will ended up with a conflict with: https://github.com/ovh-ux/ng-ovh-contacts.

I think that this dependency can be "deprecated" and stay with the previous AngularJS module name.

Is that make sense to you?

**Comment by cbourgois** [comment]
`@ovh-ux/ng-ovh-contact` !== `@ovh-ux/ng-ovh-contacts`

**Comment by antleblanc** [comment]
Please use the `Re-request review` button in that case to amend the previous requested changes.

**Review by antleblanc — CHANGES_REQUESTED** [review]
After installing the branch `refactor/rollup` I got this following diff:

```diff
diff --git a/yarn.lock b/yarn.lock
index a765bf060bdc..8d85975ffd4f 100644
--- a/yarn.lock
+++ b/yarn.lock
@@ -4431,11 +4431,6 @@ osenv@0, osenv@^0.1.4:
     os-homedir "^1.0.0"
     os-tmpdir "^1.0.0"
 
-ovh-manager-webfont@^1.2.0:
-  version "1.2.0"
-  resolved "https://registry.yarnpkg.com/ovh-manager-webfont/-/ovh-manager-webfont-1.2.0.tgz#e59e89e6248b7bf97b204bb18521b2f5bd347961"
-  integrity sha512-J2ticTEGQEKbiE0NpI8yaiGmVFrbyllonUHcIsfgaEhY1gl9cug0ktiJGTc40sVFwlygsNYoM360xZvJkeX7lw==
-
```

In fact `ovh-manager-webfont` is defined as a peer dependency so it souldn't be present in the `yarn.lock` file.

Are you able to reproduce this?

**Comment by cbourgois** [comment]
Yes, and I'm currently fixing another issue, it's why I don't re-request a review...

File: package.json [hunk 1]
```diff
@@ -1,51 +1,68 @@
 {
-  "name": "ovh-angular-contact",
+  "name": "@ovh-ux/ng-ovh-contact",
```
Comment (antleblanc) [inline package.json:2]: I think we can already replace all `ovh-angular-contact` by `ng-ovh-contact` (due to the renaming of this repository).

Is that make sense for you to do bit before merging it?

File: src/constants.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  'birthZip',
+  'birthCountry',
+  'nationality',
+  'nationalIdentificationNumber',
+  'companyNationalIdentificationNumber',
+  'emailConfirmation',
+];
+
+// eslint-disable-next-line no-useless-escape
+export const EMAIL_REGEX = "^[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]*[a-zA-Z0-9])?\.)+[a-zA-Z0-9]{2}(?:[a-zA-Z0-9-]*[a-zA-Z0-9])?$";
```
Comment (antleblanc) [inline src/constants.js]:
```suggestion
export const EMAIL_REGEX = "^[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+(?:.[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]*[a-zA-Z0-9])?.)+[a-zA-Z0-9]{2}(?:[a-zA-Z0-9-]*[a-zA-Z0-9])?$";
```

In that way we can get rid of the ESLint comment.

File: src/provider.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+import map from 'lodash/map';
+import reject from 'lodash/reject';
+import set from 'lodash/set';
+import snakeCase from 'lodash/snakeCase';
+import some from 'lodash/some';
+
+export default function () {
+  const self = this;
+
+  self.$get = function $get(
```
Comment (antleblanc) [inline src/provider.js]:
```suggestion
  self.$get = /* @ngInject */ function $get(
```

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  EMAIL_REGEX,
+} from './constants';
+
+import './index.less';
+
+const moduleName = 'ngOvhContact';
+
+angular
+  .module(moduleName, [
+    'internationalPhoneNumber',
```
Comment (antleblanc) [inline src/index.js:29]: You defined `international-phone-number` as a peer dependency but shouldn't be also imported?

**Review by antleblanc — CHANGES_REQUESTED** [review]
:tada: Well done! :tada: 

Noticed minor things that could be changed before merging it:
- :eyes: missing a `@ngInject` from the provider.
- :eyes: missing import statement for `international-phone-number`.
- :memo: previous package name references could be changed into the `package.json` file.

Regarding the translations files, you still refer to the XML (which is OK until we migrate this component into the monorepo) but maybe we can remove `Messages_fr_FR.json` file. Isn't it?

Thanks!

**Review by FredericEspiau — APPROVED** [review]

File: src/index.js (hunk 4 above)
Comment (cbourgois) [reply src/index.js:29]: Should be imported with `script-loader` in parent app.

File: package.json (hunk 1 above)
Comment (cbourgois) [reply package.json:2]: Done ;)

**Review by antleblanc — APPROVED** [review]

**Pull request merged**